### PR TITLE
increment actions/checkout from v2 to v4

### PR DESCRIPTION
**Pull request by iambumblehead**
https://github.com/actions/checkout/releases/tag/v3.0.0
https://github.com/actions/checkout/releases/tag/v4.0.0

### CODE REVIEW

**Comment by iambumblehead** [comment]
Hello, I've started using bash_unit here and it seems great https://github.com/iambumblehead/render-thumb-for.sh/pull/30

The main reason I forked bash_unit is that I wanted to submit a change to the README, however the README uses a format I'm not familiar with and sadly I don't want to spend much time with it.

Mainly, these are my subjective opinions,
 1. **there should be a screenshot or gif at the top of the README** bash-unit looks beautiful and its obvious when watching it run that this a good unit-testing tool with the things you need (whoever you are ☺️) I recently found that things like images can be exluded from releases and tags easily by adding something like the following to `.gitattributes`
    ```ini
    path/to/my.gif export-ignore
    ```
 2. **A quick list that links to description of supported assertions** should appear near the top, to show what assertions are supported. Similar to this this https://nodejs.org/api/assert.html. The reason is, this shows the user quickly that everything needed is available and supported here. Politely, imo: the current synopsis and description could be more useful. Most people who arrive at the README will know they are looking at a tool for unit-testing and what they want to know is: how features are supported? how involved or easy is it to set things up? how involved or easy is it to setup a pipeline at github or gitlab? which leads to the next opinion,
 3. **The README should feature yaml source code for a github action** and probably should link the project's own unit-test [action yaml file here.](https://github.com/pgrange/bash_unit/blob/master/.github/workflows/bash_unit.yml) For my own brief investigation one of the main concerns I had while comparing tools was "How could I setup a github action to uses this tool?" and if it looked maybe-difficult to me I moved on and looked for another tool. A strength of `bash_unit` is that it is easy to setup a unit-test pipeline on github that uses it.

Tldr; near the top include a gif, list of supported features and github action yaml source.

Thank you for bash_unit!

_bash_unit.yml_
```yaml
name: test bash_unit
on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

jobs:
  ubuntu:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v4
    - name: Unit testing with bash_unit
      run: |
        bash <(curl -s https://raw.githubusercontent.com/pgrange/bash_unit/master/install.sh)
        FORCE_COLOR=true ./bash_unit test/test*
```

**Comment by pgrange** [comment]
Hi @iambumblehead and thank you for this awesome comment.

I've taken the liberty to extract them in distinct GitHub issues so that we can merge this PR but not loose your insights here:
* #118 
* #119 
* #120 

They all make a lot of sense to me and it's always valuable to get such feedback from people who actually read what you write :)

FYI about the README formatting I used `adoc` because it can renders nicely on GitHub *AND* can be converted to man pages thanks to asciidoctor. I agree it's a tiny bit more verbose than markdown.
